eng/CodeAnalysis.src.globalconfig

```diff
@@ -393,6 +399,9 @@ dotnet_diagnostic.CA1849.severity = suggestion
 # CA1850: Prefer static 'HashData' method over 'ComputeHash'
 dotnet_diagnostic.CA1850.severity = warning
 
+# CA1851: Possible multiple enumerations of 'IEnumerable' collection
+dotnet_diagnostic.CA1851.severity = suggestion
```

Comment:
@Cosifne, @buyaa-n, I temporarily enabled this locally and fixed the things that made sense to fix, but I've left it as suggestion as there remain a non-trivial number of false positives, and it didn't seem worth it right now to suppress them each with pragmas.